integrations/integrations.go

```diff
@@ -0,0 +1,231 @@
+// SDK. It starts a Monomer node in-process instead of a CometBFT node.
+func StartCommandHandler(
+	svrCtx *server.Context,
+	clientCtx client.Context, //nolint:gocritic // hugeParam
+	appCreator servertypes.AppCreator,
+	inProcessConsensus bool,
+	opts server.StartCmdOptions,
+) error {
+	// We assume `inProcessConsensus` is true for now, so let's return an error if it's not.
+	if !inProcessConsensus {
+		return fmt.Errorf("in-process consensus must be enabled")
+	}
+
+	env := environment.New()
+	defer func() {
+		if err := env.Close(); err != nil {
+			svrCtx.Logger.Error("Failed to close environment", "err", err)
+		}
+	}()
+
+	svrCfg, err := serverconfig.GetConfig(svrCtx.Viper)
+	if err != nil {
+		return fmt.Errorf("get server config: %v", err)
+	}
+	if err := svrCfg.ValidateBasic(); err != nil {
+		return fmt.Errorf("validate server config: %v", err)
+	}
+
+	app, err := startApp(env, svrCtx, appCreator, opts)
+	if err != nil {
+		return fmt.Errorf("start application: %v", err)
+	}
+
+	// Would usually start a Comet node in-process here, but we replace the
+	// Comet node with a Monomer node.
+	if err := startInProcess(env, svrCtx, &clientCtx, app, opts); err != nil {
+		return fmt.Errorf("start Monomer node in-process: %v", err)
+	}
+
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM)
+	<-sigCh
+
+	return nil
+}
+
+var _ io.WriteCloser = (*fakeTraceWriter)(nil)
+
+type fakeTraceWriter struct{}
+
+func (f *fakeTraceWriter) Write([]byte) (int, error) {
+	return 0, nil
+}
+
+func (f *fakeTraceWriter) Close() error {
+	return nil
+}
+
+// See https://github.com/cosmos/cosmos-sdk/blob/7fb26685cd68a6c1d199dc270c80f49f2bfe7ace/server/start.go#L624
+func startApp(
+	env *environment.Env,
+	svrCtx *server.Context,
+	appCreator servertypes.AppCreator,
+	opts server.StartCmdOptions,
+) (app servertypes.Application, err error) {
+	db, err := opts.DBOpener(svrCtx.Config.RootDir, server.GetAppDBBackend(svrCtx.Viper))
+	if err != nil {
+		return app, fmt.Errorf("open db: %v", err)
+	}
+
+	// TODO: Check if is testnet and implement `testnetify` function
+
+	app = appCreator(svrCtx.Logger, db, &fakeTraceWriter{}, svrCtx.Viper)
+	env.Defer(func() {
+		if localErr := app.Close(); localErr != nil {
+			svrCtx.Logger.Error(localErr.Error())
+		}
+	})
+
+	return app, nil
+}
+
+// See https://github.com/cosmos/cosmos-sdk/blob/7fb26685cd68a6c1d199dc270c80f49f2bfe7ace/server/start.go#L307
+// We modify this function to start a Monomer node in-process instead of a Comet node.
+func startInProcess(
+	env *environment.Env,
+	svrCtx *server.Context,
+	clientCtx *client.Context,
+	app servertypes.Application,
+	opts server.StartCmdOptions,
+) error {
+	g, ctx := getCtx(svrCtx, true)
+
+	svrCtx.Logger.Info("Starting Monomer node in-process")
+	wrappedApp := &WrappedApplication{app}
+	_, err := startMonomerNode(wrappedApp, env, svrCtx)
+	if err != nil {
```

Comment:
nit: limit the scope of variables.
```suggestion
	if _, err := startMonomerNode(&WrappedApplication{
	        app: app,
	}, env, svrCtx); err != nil {
```